### ISSUE TITLE
Filter hook 'woocommerce_thankyou_order_received_text' is escaping HTML

### ISSUE THREAD

**Issue by YordanSoares** [woothemes/woocommerce#40352]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

After merging #39758, any HTML passed though the `woocommerce_thankyou_order_received_text` filter hook is being escaped:

### Expected behavior

Before merging #39758, it was possible to pass HTML code through this filter. This is very useful to add links or any other HTML code. This HTML code should be rendered correctly.

### Actual behavior

Any HTML passed though the 'woocommerce_thankyou_order_received_text' filter hook is being escaped:

### Steps to reproduce

1. Activate this code snippet in your site:
```php
add_filter('woocommerce_thankyou_order_received_text', 'wc_test_thank_you_link', 10, 2);
function wc_test_thank_you_link( $text, $order ) {
    $text .= '<p><a href="https://example.com">Click here</a></p>';
    return $text;
}
```
2. Place a new order and check the code displayed after "Thank you. Your order has been received.", which should look like this:
 
   ![image](https://github.com/woocommerce/woocommerce/assets/38109855/41215835-1a52-46da-9ffd-53d7f979e83b)

### WordPress Environment

WordPress address (URL): https://woocommerce-testing.local
Site address (URL): https:///woocommerce-testing.local
WC Version: 8.1.1
REST API Version: ✔ 8.1.1
WC Blocks Version: ✔ 10.9.3
Action Scheduler Version: ✔ 3.6.2
Log Directory Writable: ✔
WP Version: 6.3.1
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: –

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by tatof** [comment]
Same! please remove the esc_html tag from the template file :-)

More with same problem: https://wordpress.org/support/topic/woocommerce_thankyou_order_received_text-doesnt-render-html/#post-17070696

**Comment by lsinger** [comment]
Resolved by #40353. ✅
